src/components/App/App.js

```diff
@@ -10,23 +10,41 @@ export default class App extends React.Component {
 
   constructor(props){
     super(props);
-    this.state = { searchValue: '' };
+    this.state = { searchValue : '', resultsValue: undefined};
   }
 
   getLocals() {
     return {
       onSearchChange : this.onSearchChange,
-      searchValue : this.state.searchValue
+      searchValue : this.state.searchValue,
+      onLogoClick: this.onLogoClick,
+      resultsValue: this.state.resultsValue
     };
   }
 
   onSearchChange = value => {
-    console.log('I got the value ==>',value);
-    //no need for setState now
-    //this.setState({ searchValue : value });
+    let results;
+    //This is used for testing purpose only
+    if(value){
+      results = [
+        {'id': 1, 'name': 'reactjs_koans', 'description':'Learn basics of React.js making the tests pass'},
+        {'id': 2, 'name': 'react-hot-boilerplate', 'description':'Minimal live-editing boilerplate for your next ReactJS project'},
+        {'id': 3, 'name': 'generator-react-webpack', 'description':'Yeoman generator for ReactJS and Webpack'},
+        {'id': 4, 'name': 'generator-react-webpack', 'description':'Yeoman generator for ReactJS and Webpack'},
+        {'id': 5, 'name': 'generator-react-webpack', 'description':'Yeoman generator for ReactJS and Webpack'},
+        {'id': 6, 'name': 'generator-react-webpack', 'description':'Yeoman generator for ReactJS and Webpack'}
+      ];
+    }if( value !== 'reactjs' ){
+      results = [];
+    }
```

Comment:
suggestions (discussed offline in details):
1. don't use `let` (prefer `const`, unless in some very very VERY special case)
2. use `if`/`else` instead of multiple subsequent `if`
3. don't have `if` with overlapping branch conditions